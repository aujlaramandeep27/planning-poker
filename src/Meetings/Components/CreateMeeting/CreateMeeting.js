import React from 'react';
import {connect} from 'react-redux';
import {PostMeeting} from '../../Actions/MeetingsActions'
import {viewMeetings} from '../../../Navigation/route-actions';

export function CreateMeeting(props) {
    
    return (
        <div className="uk-container uk-text-center@m uk-margin-top uk-margin-bottom">
            <h1 className="uk-heading-small uk-heading-divider">Add Meeting</h1>
            <a href="#/meetings">
                <span uk-icon="icon: arrow-left; ratio: 3" className="uk-position-large uk-position-top-left"></span>
            </a>
            <div className="uk-align-center uk-width-1-2@m">      
                <form onSubmit={event => {PostNewMeeting(event)}}>
                    <div className="uk-margin">
                        <div className="uk-inline uk-width-1-2">
                            <span className="uk-form-icon" uk-icon="icon: user"></span>
                            <input name="meetingInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <button className="uk-button uk-button-primary uk-width-1-4">
                            CREATE
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );

    function PostNewMeeting(event) {
        event.preventDefault();
        props.postMeeting(event.target.meetingInputName.value.trim());
        props.goToMeetings();
    }
}

function mapStateToProps(state) {
    return {
        error: state.meetings.error
    }
 };
 
 function mapDispatchToProps(dispatch) {
     return {
        postMeeting: (meetingName) => dispatch(PostMeeting(meetingName)),
        goToMeetings: () => dispatch(viewMeetings()),
     }
 };
 
 export default connect(mapStateToProps, mapDispatchToProps)(CreateMeeting)
