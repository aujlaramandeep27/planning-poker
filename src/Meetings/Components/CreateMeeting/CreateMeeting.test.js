import {CreateMeeting} from "./CreateMeeting";

it('CreateMeeting renders correctly', () => {
    const props = {
        error: null,
        goToMeetings: jest.fn(),
        postMeeting: jest.fn()
    };
    expect(CreateMeeting(props)).toMatchSnapshot();
});
