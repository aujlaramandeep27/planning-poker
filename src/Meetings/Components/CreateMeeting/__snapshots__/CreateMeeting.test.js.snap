// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMeeting renders correctly 1`] = `
<div
  className="uk-container uk-text-center@m uk-margin-top uk-margin-bottom"
>
  <h1
    className="uk-heading-small uk-heading-divider"
  >
    Add Meeting
  </h1>
  <a
    href="#/meetings"
  >
    <span
      className="uk-position-large uk-position-top-left"
      uk-icon="icon: arrow-left; ratio: 3"
    />
  </a>
  <div
    className="uk-align-center uk-width-1-2@m"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="uk-margin"
      >
        <div
          className="uk-inline uk-width-1-2"
        >
          <span
            className="uk-form-icon"
            uk-icon="icon: user"
          />
          <input
            className="uk-input"
            name="meetingInputName"
            placeholder="Name"
            type="text"
          />
        </div>
        <button
          className="uk-button uk-button-primary uk-width-1-4"
        >
          CREATE
        </button>
      </div>
    </form>
  </div>
</div>
`;
